contributors/design-proposals/priority-eviction.md

```diff
@@ -0,0 +1,60 @@
+
+### Background and Motivation
+Prior to kubernetes v1.6, the [critical pod](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/rescheduling-for-critical-pods.md) annotation was introduced primarily to prevent eviction of system pods deemed critical.  In some cluster setups, a prerequisite for moving away from using this annotation is a properly designed system of priority that decreases the chance that high-priority pods are evicted, and is able to properly reschedule such pods quickly when they are evicted.  The [Kubernetes Priority](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-priority-api.md) proposal introduced this concept of priority to kubernetes v1.7.
+
+This proposal explores possible implementations for integrating priority with the kubelet process of eviction.  This integration should provide a method for cluster operators to provide greater availability to "high-priority" pods without needing to use the critical-pod annotation.  
+
+The current method for ranking pods for eviction is by QoS, then usage over requests.  This currently holds the invariant that a pod that does not exceed its requests is not evicted, since the sum of the requests on the node cannot exceed the allocatable memory on the node.
+
+### Goals
+ - Transparency.  Users should be able to understand why their pods are evicted, and what, if anything, they can do to prevent future evictions.
+ - Low Abuse.  High priority pods should not be able to intentionally, or unintentionally disrupt large numbers of well-behaved pods.
+ - Respect Priority.  Pods that have higher priority should be less likely to be evicted.
+
+The goal of this proposal is to build consensus on the general design of how priority is integrated with memory eviction.  This proposal itself will not be merged, but will result in a set of changes to the [kubelet eviction documentation](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/kubelet-eviction.md) after a broad design is settled on.
+
+### Non Goals
+The implementation of priority itself is outside the scope of this proposal, and is covered in the [Priority](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/pod-priority-api.md) proposal.  This includes mechanisms to control which pods are allowed to be given which priority levels.
+
+The scope of this design is restricted to the node, and does not make any proposals regarding cluster-level controllers.
+
+## Proposed Implementations
+This list is not expected to be exhaustive, but rather to explore options.  New options may be added if there is support for them.
+
+The following are proposals for how to rank pods when the node is under memory pressure.
```

Comment:
I think we need to consider at least three -- critical pods (BTW that's not just one-per-node pods, but also one-or-a-small-number-per-cluster pods like DNS and Heapster), pods that are in the serving path for external user requests, and opportunistic work (e.g. batch jobs with a flexible deadline). When HPA kicks in to scale up that second category, you want it to push out the third category but not the first category. (Obviously this assumes you can't or don't want to increase the number of nodes, e.g. non-virtualized on-prem environment or a cloud environment where you don't want to pay more.)